lookup.go

```diff
@@ -65,8 +65,8 @@ func makeName(name string, prefix string) (string, bool) {
 	}
 }
 
-func doLookup(g *GlobalLookupFactory, gc *GlobalConf, input <-chan interface{}, output chan<- string, metaChan chan<- routineMetadata, wg *sync.WaitGroup) error {
-	f, err := (*g).MakeRoutineFactory()
+func doLookup(g *GlobalLookupFactory, gc *GlobalConf, input <-chan interface{}, output chan<- string, metaChan chan<- routineMetadata, wg *sync.WaitGroup, threadID int) error {
+	f, err := (*g).MakeRoutineFactory(threadID)
```

Comment:
Is your thinking?

`f, err := g.MakeRoutineFactory(threadID)`

If so:
`
../lookup.go:69: g.MakeRoutineFactory undefined (type *GlobalLookupFactory is pointer to interface, not interface)
`